homeassistant/components/home_connect/sensor.py

```diff
@@ -69,12 +80,34 @@ async def async_update(self):
                     self._state = None
                 else:
                     seconds = self._sign * float(status[self._key][ATTR_VALUE])
-                    self._state = (
-                        dt_util.utcnow() + timedelta(seconds=seconds)
-                    ).isoformat()
+                    result_time = dt_util.utc_from_timestamp(
+                        status[self._key][ATTR_TIMESTAMP]
+                    ) + timedelta(seconds=seconds)
+
+                    if self._sign == 1 and result_time < dt_util.utcnow():
+                        # if the date is supposed to be in the future but we're
+                        # already past it, set state to None.
+                        self._state = None
+                    elif (
+                        # prevent 1 second jitter
+                        self._state is None
+                        or abs(
+                            (
+                                dt_util.parse_datetime(self._state) - result_time
+                            ).total_seconds()
+                        )
+                        > 1
+                    ):
+                        self._state = result_time.isoformat()
+            elif self.device_class == DEVICE_CLASS_TEMPERATURE:
+                self._state = round(status[self._key].get(ATTR_VALUE))
```

Comment:
If the key is missing we'll apply round to None and error. If we know that the key is in the dict we should use `dict[key]`.

Reply:
As I understand Home Connect API docs & code - yes, ATTR_VALUE always exists.
Fixed.